puffin_egui/src/stats.rs

```diff
@@ -65,13 +71,23 @@ pub fn ui(ui: &mut egui::Ui, options: &mut Options, frames: &[std::sync::Arc<Unp
                 ui.end_row();
 
                 for (key, stats) in &scopes {
-                    if !options.filter.include(key.id) {
+                    let Some(scope_details) = scope_infos.fetch_by_id(&key.id) else {
                         continue;
+                    };
+
+                    if !options.filter.include(&scope_details.function_name) {
+                        return;
                     }
 
                     ui.label(&key.thread_name);
-                    ui.label(key.location);
-                    ui.label(key.id);
+                    ui.label(scope_details.location());
+                    ui.label(format!("{:?}", scope_details.function_name));
+
+                    if let Some(name) = &scope_details.scope_name {
+                        ui.label(format!("{}", name));
+                    } else {
+                        ui.label(format!("-"));
```

Comment:
clippy should catch this…

```suggestion
                        ui.label("-");
```